### DISCUSSION TITLE
Add "Dependency Graph" hub

### DISCUSSION THREAD

**Discussion by rhyskoedijk** [mburumaxwell/paklo#1381] [Ideas]
@mburumaxwell, something I've been looking in to for a while since the "vNext" script is the idea of storing the Dependabot dependency list data in so it can be displayed in a DevOps hub; The hub would allow you to export a dependency graph across the organisation, project, or repository as either a CSV file or maybe in one of the well-known SBOM formats.

The purpose of the hub would be to allow users to track and export their dependency graph from a centralised location, or export it as file to stakeholders. There are many paid products that do this (including [GitHub Advanced Security for DevOps](https://azure.microsoft.com/en-us/products/devops/github-advanced-security)), but I'd like to see a free option since Dependabot is free and is already capable of solving this specific problem.

Below is a very basic prototype I did using the output from the new dependabot@2 task, which stores the dependency list data against the DevOps project properties (when configured to do so):

![image](https://github.com/user-attachments/assets/f532ecec-3c27-464e-83e3-89c3ab47c998)

Would you be open to accepting something like this if I was to finish it? Any concerns or suggestions? 
I would still need to finish V2 + unit tests first, this would come after that.

**Comment by mburumaxwell** [comment]
Yes! This would be great! Essentially a step towards having a UI

**Comment by mburumaxwell** [comment]
Solutions:
- [free extension](https://github.com/rhyskoedijk/sbom-azure-devops)
- [hosted version](https://paklo.app/compare)
- [official paid option](https://azure.microsoft.com/en-us/products/devops/github-advanced-security)
